File: html/data/races.js
const RACES = {
    "Human": {
        name: "Human",
        bonuses: [
            "+1 Strength",
            "+1 Toughness",
            "+1 Force of Will",
            "+1 Perception",
            "+1 Performance",
            "+1 Insight",
            "+1 Endurance",
            "+1 Resistance",
            "+1 Raw Power",
            "+1 Intimidation",
            "Starting Weapon: Longsword",
            "Starting Food: 1D6",
            "Starting Worthiness: 7",
            "Starting HP: 20"
        ],
        startingEquipment: {
            weapon: "Longsword",
            food: "1D6",
            gold: 0,
            worthiness: 7,
            hp: 20
        }
    },
    "Dwarf": {
        name: "Dwarf",
        bonuses: [
            "+2 Strength",
            "+2 Toughness",
            "+1 Force of Will",
            "+2 Raw Power",
            "+2 Resistance",
            "+1 Endurance",
            "Starting Weapon: Warhammer",
            "Starting Food: 1D10",
            "Starting Worthiness: 7",
            "Starting HP: 22"
        ],
        startingEquipment: {
            weapon: "Warhammer",
            food: "1D10",
            gold: 0,
            worthiness: 7,
            hp: 22
        }
    },
    "Halfling": {
        name: "Halfling",
        bonuses: [
            "+1 Strength",
            "+1 Toughness",
            "+1 Force of Will",
            "+2 Perception",
            "+2 Insight",
            "+1 Performance",
            "+1 Endurance",
            "+1 Resistance",
            "Starting Weapon: Sling",
            "Starting Food: 1D6",
            "Starting Worthiness: 8",
            "Starting HP: 14"
        ],
        startingEquipment: {
            weapon: "Sling",
            food: "1D6",
            gold: 0,
            worthiness: 8,
            hp: 14
        }
    },
    "High Elf": {
        name: "High Elf",
        bonuses: [
            "+1 Strength",
            "+1 Toughness",
            "+2 Force of Will",
            "+2 Perception",
            "+2 Insight",
            "+2 Performance",
            "+1 Endurance",
            "+1 Resistance",
            "Starting Weapon: Scimitar",
            "Starting Food: 1D10",
            "Starting Worthiness: 10",
            "Starting HP: 16"
        ],
        startingEquipment: {
            weapon: "Scimitar",
            food: "1D10",
            gold: 0,
            worthiness: 10,
            hp: 16
        }
    },
    "Moon Elf": {
        name: "Moon Elf",
        bonuses: [
            "+1 Strength",
            "+1 Toughness",
            "+1 Force of Will",
            "+2 Insight",
            "+2 Endurance",
            "+1 Perception",
            "+1 Intimidation",
            "+1 Resistance",
            "Starting Weapon: Shortsword",
            "Starting Food: 1D6",
            "Starting Worthiness: 4",
            "Starting HP: 18"
        ],
        startingEquipment: {
            weapon: "Shortsword",
            food: "1D6",
            gold: 0,
            worthiness: 4,
            hp: 18
        }
    },
    "Orc": {
        name: "Orc",
        bonuses: [
            "+2 Strength",
            "+2 Toughness",
            "+2 Raw Power",
            "+2 Intimidation",
            "+1 Endurance",
            "+1 Resistance",
            "Starting Weapon: Greatclub",
            "Starting Food: 1D6",
            "Starting Worthiness: -5",
            "Starting HP: 24"
        ],
        startingEquipment: {
            weapon: "Greatclub",
            food: "1D6",
            gold: 0,
            worthiness: -5,
            hp: 24
        }
    },
    "Troll": {
        name: "Troll",
        bonuses: [
            "+1 Strength",
            "+2 Toughness",
            "+1 Force of Will",
            "+2 Endurance",
            "+2 Resistance",
            "+1 Raw Power",
            "+1 Perception",
            "Starting Weapon: Club",
            "Starting Food: 1D6",
            "Starting Worthiness: 2",
            "Starting HP: 20"
        ],
        startingEquipment: {
            weapon: "Club",
            food: "1D6",
            gold: 0,
            worthiness: 2,
            hp: 20
        }
    }
};

File: android/app/src/main/java/nu/aedelore/app/domain/gamedata/Races.kt
package nu.aedelore.app.domain.gamedata

object Races {

    data class Race(
        val name: String,
        val bonuses: List<String>,
        val hp: Int,
        val worthiness: Int,
        val food: String
    )

    val all: List<Race> = listOf(
        Race(
            name = "Human",
            bonuses = listOf(
                "+1 Strength", "+1 Dexterity", "+1 Toughness", "+1 Intelligence",
                "+1 Wisdom", "+1 Athletics", "+1 Perception", "+1 Performance",
                "+1 Survival", "+1 Investigation"
            ),
            hp = 20,
            worthiness = 7,
            food = "1D6"
        ),
        Race(
            name = "Dwarf",
            bonuses = listOf(
                "+2 Strength", "+1 Toughness", "+1 Force of Will", "+1 Raw Power",
                "+2 History", "+2 Survival", "+1 Resistance"
            ),
            hp = 22,
            worthiness = 7,
            food = "1D10"
        ),
        Race(
            name = "Halfling",
            bonuses = listOf(
                "+2 Dexterity", "+1 Strength", "+1 Toughness", "+1 Intelligence",
                "+1 Wisdom", "+1 Acrobatics", "+1 Stealth", "+1 Sleight of Hand",
                "+1 Perception"
            ),
            hp = 14,
            worthiness = 8,
            food = "1D6"
        ),
        Race(
            name = "High Elf",
            bonuses = listOf(
                "+1 Dexterity", "+2 Intelligence", "+2 Wisdom", "+1 Acrobatics",
                "+2 History", "+2 Nature", "+1 Medicine", "+1 Perception"
            ),
            hp = 16,
            worthiness = 10,
            food = "1D10"
        ),
        Race(
            name = "Moon Elf",
            bonuses = listOf(
                "+2 Dexterity", "+1 Wisdom", "+2 Acrobatics", "+1 Stealth",
                "+1 Unarmed", "+1 Endurance", "+1 Animal Handling", "+1 Insight"
            ),
            hp = 18,
            worthiness = 4,
            food = "1D6"
        ),
        Race(
            name = "Orc",
            bonuses = listOf(
                "+2 Strength", "+2 Toughness", "+2 Raw Power", "+1 Endurance",
                "+1 Unarmed", "+2 Intimidation"
            ),
            hp = 24,
            worthiness = -5,
            food = "1D6"
        ),
        Race(
            name = "Troll",
            bonuses = listOf(
                "+1 Dexterity", "+2 Toughness", "+2 Athletics", "+1 Endurance",
                "+2 Resistance", "+2 Survival"
            ),
            hp = 20,
            worthiness = 2,
            food = "1D6"
        )
    )

    fun byName(name: String): Race? = all.find {
        it.name.equals(name, ignoreCase = true)
    }
}
